lib/chef/provider/aws_cloudsearch_domain.rb

```diff
@@ -0,0 +1,126 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsCloudsearchDomain < Chef::Provisioning::AWSDriver::AWSProvider
+
+  def create_aws_object
+    converge_by "Creating new CloudSearch domain #{new_resource.domain_name}" do
+      create_domain
+    end
+    # multi_az defaults to false on creation
+    # so we only need to hit the API if it is true
+    if new_resource.multi_az
```

Comment:
Lets DRY this up by just calling `update_aws_object(instance)`.  The `instance` should be returned from the `create_domain` method.